### PR TITLE
feat: add either interface on restclients with all response status codes

### PR DESCRIPTION
**Pull request by pevisscher**

### CODE REVIEW

**Comment by pevisscher** [comment]
Current dependencies on/for this PR:
* main
  * **PR #5** <a href="https://app.graphite.dev/github/pr/skyleague/therefore/5" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/skyleague/therefore/5?utm_source=stack-comment).

File: src/lib/primitives/jsonschema/jsonschema.ts [hunk 1]
```diff
@@ -231,10 +231,11 @@ async function walkJsonschema({
 
     if (isArray(child)) {
         // tuple json schema definition
-        return $tuple(
-            await asyncCollect(child.map((c) => walkJsonschema({ node: c, visitor, childProperty: undefined, context }))),
-            annotate(node, context)
-        )
+        const schemas: CstSubNode[] = []
+        for await (const c of child) {
```
Comment (MickVanDuijn) [inline src/lib/primitives/jsonschema/jsonschema.ts:235]: decided not to use `axioms` in the end?

File: test/petstore.spec.ts [hunk 2]
```diff
@@ -17,6 +17,6 @@ describe('updatePet', () => {
         const updatedPet: Pet = { name: 'FooPet', photoUrls: ['example.com'] }
         nock('http://www.example.com:80').put('/pet').reply(200, updatedPet)
 
-        expect(await client.updatePet({ body: updatedPet })).toEqual(updatedPet)
+        expect((await client.updatePet({ body: updatedPet })).right).toEqual(updatedPet)
```
Comment (MickVanDuijn) [inline test/petstore.spec.ts:20]: how come `right` is not triggering an error here? :thinking_face: there is no `isRight` guard around it

**Review by MickVanDuijn — APPROVED** [review]

**Comment by pevisscher** [comment]
**Graphite Merge Job**

Current status: 🔄 Waiting for checks to pass

This pull request is currently waiting for CI and mergeability checks to pass before being merged.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/skyleague/therefore/5?utm_source=stack-comment).


Job Reference: f0penGwW85vorg9X6nFC

**Pull request merged**